pymeasure/instruments/hcp/tc038d.py

```diff
@@ -54,90 +61,94 @@ class TC038D(Instrument):
 
     byteMode = 4
 
-    functions = {'read': 0x03, 'writeMultiple': 0x10,
-                 'writeSingle': 0x06, 'echo': 0x08}
-
     def __init__(self, adapter, name="TC038D", address=1, timeout=1000,
                  **kwargs):
         """Initialize the device."""
         super().__init__(adapter, name, timeout=timeout, **kwargs)
         self.address = address
 
-    def readRegister(self, address, count=1):
-        """Read count variables from start address on."""
-        # Count has to be double the number of elements in 4-byte-mode.
-        count *= self.byteMode // 2
-        data = [self.address]
-        data.append(self.functions['read'])  # function code
-        data += [address >> 8, address & 0xFF]  # 2B address
-        data += [count >> 8, count & 0xFF]  # 2B number of elements
-        data += CRC16(data)
-        self.write_bytes(bytes(data))
-        # Slave address, function, length
-        got = self.read_bytes(3)
-        if got[1] == self.functions['read']:
-            length = got[2]
-            # data length, 2 Byte CRC
-            read = self.read_bytes(length + 2)
-            if read[-2:] != bytes(CRC16(got + read[:-2])):
-                raise ConnectionError("Response CRC does not match.")
-            return read[:-2]
-        else:  # an error occurred
-            end = self.read_bytes(2)  # empty the buffer
-            if got[2] == 0x02:
-                raise ValueError(f"The read start address {address} is invalid.")
-            if got[2] == 0x03:
-                raise ValueError(f"The number of elements {count} exceeds the allowed range.")
-            raise ConnectionError(f"Unknown read error. Received: {got} {end}")
-
-    def writeMultiple(self, address, values):
-        """Write multiple variables."""
-        data = [self.address]
-        data.append(self.functions['writeMultiple'])  # function code
-        data += [address >> 8, address & 0xFF]  # 2B address
-        if isinstance(values, int):
-            data += [0x0, self.byteMode // 2]  # 2B number of elements
-            data.append(self.byteMode)  # 1B number of write data
-            for i in range(self.byteMode - 1, -1, -1):
-                data.append(values >> i * 8 & 0xFF)
-        elif isinstance(values, collections.abc.Sequence):
+    def write(self, command):
+        """Write a command to the device.
+
+        :param str command: comma separated string of:
+            - the function: read ('R') or write ('W') or 'echo',
+            - the address to write to (e.g. '0x106' or '262'),
+            - the values (comma separated) to write
+            - or the number of elements to read (defaults to 1).
+        """
+        function, address, *values = command.split(",")
+        data = [self.address]  # 1B device address
+        data.append(Functions[function].value)  # 1B function code
```

Comment:
IIRC, if you had `class Functions(IntEnum)` instead, I think this becomes easier because you could just do `Functions[function]`, as you then get the integer when you need it?

Reply:
I never used enums before, but the official documentation is quite good, therefore I noticed IntEnums, but they did not seem to offer any benefit here.
I tried it with an IntEnum, and it behaves the same like Enum (only comparisons work differently).

Reply:
The key difference is that an IntEnum behaves like an int.
> IntEnum is the same as Enum, but its members are also integers and can be used anywhere that an integer can be used. If any integer operation is performed with an IntEnum member, the resulting value loses its enumeration status.

This enables you (mostly) to transparently use them in a context where you are working with integers, but also want what enums offer, like here. :-)

Reply:
Oh great, I push the changes immediately.

Reply:
Cool, we are done, then! 😊